app/scripts/services/hpa.js

```diff
@@ -146,15 +168,27 @@ angular.module("openshiftConsole")
         return $q.when([]);
       }
       return MetricsService.isAvailable().then(function(metricsAvailable) {
+        var v2HPAWarning = hasV2HPAScaler(scaleTarget, hpaResources);
         return _.compact([
           hasMetricsAvailableWarning(metricsAvailable),
-          hasCPURequestWarning(scaleTarget, limitRanges, project),
+          v2HPAWarning ? v2HPAWarning : hasCPURequestWarning(scaleTarget, limitRanges, project),
           hasCompetingAutoscalersWarning(hpaResources),
           hasCompetingDCAndAutoscalerWarning(scaleTarget, hpaResources)
         ]);
       });
     };
 
+    // Get HPA warnings relevant to a Resource list page.
+    //
+    // This method returns the same set as the above getHPAWarnings but
+    // filters out the warnings that don't need to be fully surfaced to
+    // each individual resource.
+    var getHPAWarningsForResource = function(scaleTarget, hpaResources, limitRanges, project) {
```

Comment:
Are we filtering out the v2 warnings from all places we show them? For instance, the deployment donut on the overview?

I think it would be safer to opt in to the v2 editor warnings instead of always including editor warnings and having to filter them out. There's only one context where we need the extra warning: the editor. And that context performs the check in a different way (`isUnsupportedAPI()`).

Reply:
Ah, I didn't check the donut, thanks, this is problematic:

![screen shot 2018-02-01 at 3 22 10 pm](https://user-images.githubusercontent.com/280512/35701392-d6f8d204-0763-11e8-97b8-e017b6078e6d.png)
![screen shot 2018-02-01 at 3 22 32 pm](https://user-images.githubusercontent.com/280512/35701393-d707941a-0763-11e8-95f1-69962df8eda5.png)

I think I may still need to do the `hasV2HPAScalerWarning` check in `getHPAWarnings`, but then just drop both `v2` && the `CPURequest` warning.